src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzerTests.cs

```diff
@@ -291,6 +291,65 @@ public override void Initialize(AnalysisContext context)
                 GetCSharpRS1019ExpectedDiagnostic(38, 34, "DuplicateDiagnosticId", "MyAnalyzer"));
         }
 
+        [Fact]
+        public async Task RS1017_RS1019_CSharp_VerifyDiagnostic_CreateHelper()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+class MyAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly string NonConstantDiagnosticId = ""NonConstantDiagnosticId"";
+    private static LocalizableResourceString dummyLocalizableTitle = null;
+
+    private static readonly DiagnosticDescriptor descriptor =
+        DiagnosticDescriptorHelper.Create(NonConstantDiagnosticId, dummyLocalizableTitle, ""MyDiagnosticMessage"", ""MyDiagnosticCategory"");
+
+    private static readonly DiagnosticDescriptor descriptor2 =
+        DiagnosticDescriptorHelper.Create(""DuplicateDiagnosticId"", dummyLocalizableTitle, ""MyDiagnosticMessage"", ""MyDiagnosticCategory"");
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+    {
+        get
+        {
+            return ImmutableArray.Create(descriptor, descriptor2);
+        }
+    }
+
+    public override void Initialize(AnalysisContext context)
+    {
+    }
+}
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+class MyAnalyzer2 : DiagnosticAnalyzer
+{
+    private static LocalizableString dummyLocalizableTitle = null;
+
+    private static readonly DiagnosticDescriptor descriptor =
+        DiagnosticDescriptorHelper.Create(""DuplicateDiagnosticId"", dummyLocalizableTitle, ""MyDiagnosticMessage"", ""MyDiagnosticCategory"");
+
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+    {
+        get
+        {
+            return ImmutableArray.Create(descriptor);
+        }
+    }
+
+    public override void Initialize(AnalysisContext context)
+    {
+    }
+}" + CSharpDiagnosticDescriptorCreationHelper,
+                GetCSharpRS1017ExpectedDiagnostic(14, 43, "descriptor"),
+                GetCSharpRS1019ExpectedDiagnostic(38, 43, "DuplicateDiagnosticId", "MyAnalyzer"));
```

Comment:
💡 Consider using markup syntax so it's easier to see the full span where these are getting reported.

Reply:
I have kept it consistent with the existing tests in this file. We can clean up the entire file in a follow-up PR.